staging/src/k8s.io/component-base/logs/logs.go

```diff
@@ -20,25 +20,76 @@ import (
 	"flag"
 	"fmt"
 	"log"
+	"sync"
 	"time"
 
 	"github.com/spf13/pflag"
-	"k8s.io/apimachinery/pkg/util/wait"
 	"k8s.io/klog/v2"
 )
 
 const logFlushFreqFlagName = "log-flush-frequency"
 
-var logFlushFreq = pflag.Duration(logFlushFreqFlagName, 5*time.Second, "Maximum number of seconds between log flushes")
+var (
+	logFlushFreqVar *pflag.Flag
+
+	// logFlushPeriod is the amount of time between log flushes. It might
+	// change over time while log flushing has already started, for example
+	// because flag parsing is done (again) later. Therefore access to it
+	// has to be protected with a mutex.
+	logFlushPeriod = &guardedDuration{
+		duration: 5 * time.Second,
+	}
+)
+
+type guardedDuration struct {
+	mutex    sync.Mutex
+	duration time.Duration
+}
+
+func (g *guardedDuration) String() string {
+	g.mutex.Lock()
+	defer g.mutex.Unlock()
+	return g.duration.String()
+}
+
+func (g *guardedDuration) Set(value string) error {
```

Comment:
Does this implement a particular interface?  Or why are these functions needed?

Reply:
They implement flag.Value resp. pflag.Value methods. There's a type assertion further down (`var _ pflag.Value = &guardedDuration{}`). Shall I call that out already as method comment ("Set implements flag.Value.Set", etc.)?

Reply:
Ah.  I just missed it.